modules/margin-protocol/src/lib.rs

```diff
@@ -503,11 +490,27 @@ impl<T: Trait> Module<T> {
 		NextPositionId::mutate(|id| *id += 1);
 
 		<Positions<T>>::insert(id, position);
-		<PositionsByTrader<T>>::mutate(who, pool, |ids| ids.push(id));
-		PositionsByPool::mutate(pool, pair, |ids| ids.push(id));
+		<PositionsByTrader<T>>::insert(who, (pool, id), ());
+		PositionsByPool::insert(pool, (pair, id), ());
 
 		Ok(())
 	}
+
+	fn _ensure_can_open_more_position(who: &T::AccountId, pool: LiquidityPoolId, pair: TradingPair) -> DispatchResult {
+		let count = PositionsByPool::iter(pool).filter(|((p, _), _)| *p == pair).count();
```

Comment:
I am sure there are some low level magic to do counting without iterate everything...

Something to be revisited later.

Reply:
https://github.com/laminar-protocol/laminar-chain/issues/153